/hardware_node:
  ros__parameters:
    motor_names:
    - pan_motor
    - tilt_motor
    - left_wheel_motor
    - right_wheel_motor
    - fire_solenoid
    left_wheel_motor:
      brake_mode: false
      current_lim:
        abs_lim: 20.0
        enable: true
        lim_trigger: 15.0
        time_window: 0.5
      follower: -1
      id: 3
      inverted: false
      # Tuned
      pid:
        izone: 150.0
        kd: 0.0
        kf: 0.051
        ki: 0.0010
        kp: 1.0
      type: TalonFx
      vcomp: 11.0
    pan_motor:
      brake_mode: true
      current_lim:
        abs_lim: 20.0
        enable: true
        lim_trigger: 15.0
        time_window: 0.5
      follower: -1
      id: 2
      inverted: false
      motion_magic:
        accel: 40.0
        cruise_vel: 40.0
        smoothing: 1
      pid:
        izone: 0.0
        kd: 6.0
        kf: 0.0
        ki: 0.0
        kp: 1.5
      type: TalonFx
      vcomp: 11.0
    right_wheel_motor:
      brake_mode: false
      current_lim:
        abs_lim: 20.0
        enable: true
        lim_trigger: 15.0
        time_window: 0.5
      follower: -1
      id: 4
      inverted: true
      # Tuned
      pid:
        izone: 150.0
        kd: 0.0
        kf: 0.051
        ki: 0.0010
        kp: 1.0
      type: TalonFx
      vcomp: 11.0
    # zero point at 23.275
    tilt_motor:
      brake_mode: true
      current_lim:
        abs_lim: 40.0
        enable: true
        lim_trigger: 25.0
        time_window: 2.0
      follower: -1
      id: 1
      inverted: false
      motion_magic:
        accel: 30.0
        cruise_vel: 30.0
        smoothing: 1
      pid:
        izone: 0.0
        kd: 0.04
        kf: 0.0
        ki: 0.0
        kp: 1.4
      type: TalonFx
      vcomp: 11.0
    fire_solenoid:
      brake_mode: false
      current_lim:
        abs_lim: 20.0
        enable: true
        lim_trigger: 5.0
        time_window: 2.0
      follower: -1
      id: 5
      inverted: false
      pid:
        izone: 0.0
        kd: 0.0
        kf: 0.0
        ki: 0.0
        kp: 0.5
      type: TalonSRx
      vcomp: 11.0
